platform/migration/platform-migratable-util/src/main/java/org/codice/ddf/migration/util/MigratableUtil.java

```diff
@@ -0,0 +1,275 @@
+     */
+    public void copyFile(@NotNull Path sourceFile, @NotNull Path exportDirectory,
+            @NotNull Collection<MigrationWarning> warnings) throws MigrationException {
+        copy(sourceFile,
+                exportDirectory,
+                warnings,
+                () -> isSourceMigratable(sourceFile,
+                        () -> new PathMigrationWarning(sourceFile, "is absolute"),
+                        () -> new PathMigrationWarning(sourceFile, "contains a symbolic link"),
+                        () -> new PathMigrationWarning(sourceFile,
+                                String.format("is outside [%s]", ddfHome)),
+                        warnings));
+    }
+
+    /**
+     * Recursively copies a directory to a destination directory. The directory to
+     * copy must be a relative path under {@code ddf.home}, and its path must not contain any
+     * symbolic link, otherwise the directory will not be copied and a {@link MigrationWarning}
+     * will be returned.
+     *
+     * @param source          relative path to the directory to copy
+     * @param exportDirectory root directory where the file will be copied. If the file'
+     *                        relative path included any directories, those will be re-created
+     *                        under this directory.
+     * @param warnings        any warnings generated during this operation (e.g., source file outside
+     *                        of {@code ddf.home}) will be added to this collection
+     * @throws MigrationException thrown if file system error prevents the directory to be copied
+     */
+    public void copyDirectory(@NotNull Path source, @NotNull Path exportDirectory,
+            @NotNull Collection<MigrationWarning> warnings) throws MigrationException {
+        copyFile(source, exportDirectory, warnings);
+    }
+
+    /**
+     * Copies a file, whose path is taken from a {@link System} property value, to a destination
+     * directory. The file to copy must be a relative path under {@code ddf.home}, and its path
+     * must not contain any symbolic link, otherwise the file will not be copied and a
+     * {@link MigrationWarning} will be returned.
+     *
+     * @param systemProperty  name of the {@link System} property that contains the path to the
+     *                        source file
+     * @param exportDirectory root directory where the file will be copied. If the file'
+     *                        relative path included any directories, those will be re-created
+     *                        under this directory.
+     * @param warnings        any warnings generated during this operation (e.g., source file outside
+     *                        of {@code ddf.home}) will be added to this collection
+     */
+    public void copyFileFromSystemPropertyValue(@NotNull String systemProperty,
+            @NotNull Path exportDirectory, @NotNull Collection<MigrationWarning> warnings) {
```

Comment:
Added `MigratableException` on all the methods that didn't have it.

Regarding the `notNull()` checks, the private `copy` method already does that and I don't think we want them twice.